### PR TITLE
Change the way we store and retrieve content source

### PR DESCRIPTION
**Pull request by LATaylor-guardian**
Change the way we store and retrieve content source by making it contextual to its environment (LIVE|DRAFT|CODE|PROD)

### CODE REVIEW

File: app/com/gu/floodgate/contentsource/ContentSource.scala [hunk 1]
```diff
@@ -13,16 +13,29 @@ case class ContentSourcesResponse(contentSources: Seq[ContentSource])
...
 @JsonFormat
-case class ContentSource(id: String = UUID.randomUUID().toString,
+case class ContentSource(id: String,
+  appName: String,
+  description: String,
+  reindexEndpoint: String,
+  environment: String)
+
+@JsonFormat
+case class ContentSourceForUpdates(
```
Comment (cb372) [inline app/com/gu/floodgate/contentsource/ContentSource.scala]: I find this naming a little confusing. What's the difference between `ContentSource` and `ContentSourceForUpdates`?

File: app/com/gu/floodgate/reindex/ProgressTrackerController.scala [hunk 2]
```diff
@@ -18,7 +18,7 @@ object ProgressTrackerController {
 
 class ProgressTrackerController(ws: WSAPI, runningJobService: RunningJobService, jobHistoryService: JobHistoryService) extends Actor with ActorLogging with StrictLogging {
 
-  private var runningTrackers: Map[String, ActorRef] = Map.empty[String, ActorRef]
+  private var runningTrackers: Map[ContentSource, ActorRef] = Map.empty[ContentSource, ActorRef]
```
Comment (cb372) [inline app/com/gu/floodgate/reindex/ProgressTrackerController.scala]: Using the `ContentSource` itself as the key is potentially a little dangerous. e.g. if somebody updates the content source's description during a reindex, you will lose track of it in this map.

Maybe generate a unique ID based on the ID and the environment? e.g.

```scala
case class ContentSource(...) {
   def uniqueId: String = s"$id-$environment"
}

**Comment by cb372** [comment]
Looks :+1: apart from comments. I totally skipped over the React stuff :)

File: app/com/gu/floodgate/contentsource/ContentSource.scala (hunk 1 above)
Comment (LATaylor-guardian) [inline app/com/gu/floodgate/contentsource/ContentSource.scala]: The difference is that when we update we don't allow them to update the environment and therefore this is not included.

**Pull request merged**